runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java

```diff
@@ -231,6 +235,40 @@ public void onNavigationButtonRemoved(final NavigationButtonRemoved event)
 		});
 	}
 
+	@Subscribe
+	public void onGameTick(final GameTick event)
```

Comment:
Why is this done every game tick instead of on game state changed?

Reply:
Because on login you dont have username yet

Reply:
Huh.. Would've thought you could get it once game state changed to LOGGED_IN. Alright then. :man_shrugging: